### PR TITLE
feat: [DEL-4218]: break dependency of delegate scripts jre on manager

### PR DESCRIPTION
**Pull request by arpit3107**
## Describe your changes
delegate and watcher should not depend on manager to get the jreConfigs (version,jrePath etc..).
These configs are now hardcoded in delegate and watcher scripts.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35777)

### CODE REVIEW

**Comment by jennyjames21** [comment]
*[`260-delegate/container/scripts/delegate.sh` line 238 at r2](https://reviewable.io/reviews/harness/harness-core/35777#-N85KQLZ2ivCu3N2Ib_m:-N85KQLZ2ivCu3N2Ib_n:b-hfpv1m) ([raw file](https://github.com/harness/harness-core/blob/9c739fe6e360b3d2a47afe5ed11c9a47eaa54203/260-delegate/container/scripts/delegate.sh#L238)):*
> ```Shell
> export CAPSULE_CACHE_DIR="$DIR/.cache"
> 
> #kya karein
> ```

remove?

**Comment by jennyjames21** [comment]
*[`400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java` line 1714 at r2](https://reviewable.io/reviews/harness/harness-core/35777#-N85KcXdAUxP9Qck5q58:-N85KcXdAUxP9Qck5q59:b-bvlk3b) ([raw file](https://github.com/harness/harness-core/blob/9c739fe6e360b3d2a47afe5ed11c9a47eaa54203/400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java#L1714)):*
> ```Java
>   private String getTargetJreVersion(final String accountId) {
>     final String jreVersion = mainConfiguration.getMigrateToJre();
>     JreConfig jreConfig = mainConfiguration.getJreConfigs().get(jreVersion);
> ```

can jreconfig be null for any environment?

**Comment by jennyjames21** [comment]
*[`400-rest/src/main/resources/delegatetemplates/common/delegate-environment.ftl` line 21 at r2](https://reviewable.io/reviews/harness/harness-core/35777#-N85KqQ10TrFKmmPKVBi:-N85KqQ10TrFKmmPKVBj:bogqj3k) ([raw file](https://github.com/harness/harness-core/blob/9c739fe6e360b3d2a47afe5ed11c9a47eaa54203/400-rest/src/main/resources/delegatetemplates/common/delegate-environment.ftl#L21)):*
> ```FreeMarker
>         </#if>
>         - name: JRE_VERSION
>           value: 11.0.14
> ```

is it good idea to hardcode? On next update we have to change to multiple places then.

**Review by arpit3107 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35777)* status: 12 file(s) to review. 3. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`260-delegate/container/scripts/delegate.sh` line 238 at r2](https://reviewable.io/reviews/harness/harness-core/35777#-N85KQLZ2ivCu3N2Ib_m:-N87AeZyDmoIneoKNisP:b-31jl4o) ([raw file](https://github.com/harness/harness-core/blob/9c739fe6e360b3d2a47afe5ed11c9a47eaa54203/260-delegate/container/scripts/delegate.sh#L238)):*
<details><summary><i>Previously, jennyjames21 (Jenny James) wrote…</i></summary><blockquote>

remove?
</blockquote></details>

Actually i wanted to ask one question: What should we do for this alpn-boot-jar . I am not sure how this works, hence wanted to ask whether should we remove this or not.
___
*[`400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java` line 1714 at r2](https://reviewable.io/reviews/harness/harness-core/35777#-N85KcXdAUxP9Qck5q58:-N87BE3d5M1qWTjtU3sw:b-nnmrbt) ([raw file](https://github.com/harness/harness-core/blob/9c739fe6e360b3d2a47afe5ed11c9a47eaa54203/400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java#L1714)):*
<details><summary><i>Previously, jennyjames21 (Jenny James) wrote…</i></summary><blockquote>

can jreconfig be null for any environment?
</blockquote></details>

I guess no, i have just removed one private method which used to fetch JREConfigs, so I have removed that method and modified it for fetching version directly.
___
*[`400-rest/src/main/resources/delegatetemplates/common/delegate-environment.ftl` line 21 at r2](https://reviewable.io/reviews/harness/harness-core/35777#-N85KqQ10TrFKmmPKVBi:-N87BkxZ6pKYM8WxYUmJ:b7skzey) ([raw file](https://github.com/harness/harness-core/blob/9c739fe6e360b3d2a47afe5ed11c9a47eaa54203/400-rest/src/main/resources/delegatetemplates/common/delegate-environment.ftl#L21)):*
<details><summary><i>Previously, jennyjames21 (Jenny James) wrote…</i></summary><blockquote>

is it good idea to hardcode? On next update we have to change to multiple places then.
</blockquote></details>

Yes we are good with updating at all the places, because we don't want to depend on manager for any jreConfigs.

**Review by gaurav-jain1 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35777)* status: 12 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`260-delegate/container/scripts/delegate.sh` line 69 at r2](https://reviewable.io/reviews/harness/harness-core/35777#-N88Gm-BBhX7GoHQ8zpL:-N88Gm-BBhX7GoHQ8zpM:b-pi9d9n) ([raw file](https://github.com/harness/harness-core/blob/9c739fe6e360b3d2a47afe5ed11c9a47eaa54203/260-delegate/container/scripts/delegate.sh#L69)):*
> ```Shell
> JRE_BINARY=$JRE_DIR/bin/java
> 
> #iska kya karein
> ```

remove this



_Code quote:_
```Shell
#iska kya karein
```
___
*[`260-delegate/container/scripts/start.sh` line 58 at r2](https://reviewable.io/reviews/harness/harness-core/35777#-N88GujX4EwvLqhocrMe:-N88GujX4EwvLqhocrMf:b-ys9vwp) ([raw file](https://github.com/harness/harness-core/blob/9c739fe6e360b3d2a47afe5ed11c9a47eaa54203/260-delegate/container/scripts/start.sh#L58)):*
> ```Shell
> fi
> 
> #Question? when is this used?
> ```

I dont think we use this file anywhere, consider deleting it.



_Code quote:_
```Shell
#Question? when is this used?
```
___
*[`400-rest/src/main/resources/delegatetemplates/common.sh.ftl` line 51 at r2](https://reviewable.io/reviews/harness/harness-core/35777#-N88IT297yddPH70Tcbx:-N88IT297yddPH70Tcby:b-v183pw) ([raw file](https://github.com/harness/harness-core/blob/9c739fe6e360b3d2a47afe5ed11c9a47eaa54203/400-rest/src/main/resources/delegatetemplates/common.sh.ftl#L51)):*
> ```FreeMarker
> DELEGATE_STORAGE_URL=${delegateStorageUrl}
> 
> <#if useCdn == "true">
> ```

prefer not to use usecdn



_Code quote:_
```FreeMarker
useCdn
```
___
*[`dockerization/delegate/scripts/start.sh` line 182 at r2](https://reviewable.io/reviews/harness/harness-core/35777#-N88IujR-C34-OWM27y9:-N88IujR-C34-OWM27yA:bmx1rmi) ([raw file](https://github.com/harness/harness-core/blob/9c739fe6e360b3d2a47afe5ed11c9a47eaa54203/dockerization/delegate/scripts/start.sh#L182)):*
> ```Shell
> fi
> 
> #should we change this as well? (IMO: YES)
> ```

Yes this is the only files needs to be changed.  I guess .ftl files don't need to be changed, you can use ${jreVersion} from manager.



_Code quote:_
```Shell
#should we change this as well? (IMO: YES)
```

File: 260-delegate/container/scripts/delegate.sh [hunk 1]
```diff
@@ -234,6 +235,7 @@ rm -f -- *.bak
 export HOSTNAME
 export CAPSULE_CACHE_DIR="$DIR/.cache"
 
+#kya karein
```
Comment (bharatgoelharness) [inline 260-delegate/container/scripts/delegate.sh:238]: lets delete this :)

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/35777)

**Review by mdmilic — COMMENTED** [review]
> Reviewed 7 of 12 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35777)* status: Some of the checks are too old. 1 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`260-delegate/container/scripts/delegate.sh` line 238 at r2](https://reviewable.io/reviews/harness/harness-core/35777#-N85KQLZ2ivCu3N2Ib_m:-N8j3hFY7kHHkeyHt13k:bjojdtk) ([raw file](https://github.com/harness/harness-core/blob/9c739fe6e360b3d2a47afe5ed11c9a47eaa54203/260-delegate/container/scripts/delegate.sh#L238)):*
<details><summary><i>Previously, arpit3107 (Arpit Tiwari) wrote…</i></summary><blockquote>

Actually i wanted to ask one question: What should we do for this alpn-boot-jar . I am not sure how this works, hence wanted to ask whether should we remove this or not.
</blockquote></details>

alpn is only needed for JDK8, with JDK11 we don't need it any more and in fact JVM will fail it has it on the class path 
___
*[`400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java` line 1490 at r2](https://reviewable.io/reviews/harness/harness-core/35777#-N8j4oZUEXE-80vpJkdi:-N8j4oZUEXE-80vpJkdj:b-y5vt51) ([raw file](https://github.com/harness/harness-core/blob/a713bc5284f258154a20a52077499bb2a9cef001/400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java#L1490)):*
> ```Java
>       Preconditions.checkNotNull(jreConfig, "jreConfig cannot be null");
> 
>       params.put(JRE_VERSION_KEY, jreConfig.getVersion());
> ```

I would keep this. It's not really great but we don't have a better way to migrate legacy delegates right now. We might need to migrate to 11.0.15 at some point.

**Pull request closed without merging**